counsel.el

```diff
@@ -1322,34 +1322,38 @@ files in a project.")
        (setq cmd counsel-git-grep-cmd-default)))
     (cons proj cmd)))
 
-(defun counsel--call (&rest program-and-args)
-  (let ((stderr-file (make-temp-file "counsel-call-stderr")))
+(defun counsel--call (&rest command)
+  "Synchronously call COMMAND and return its output as a string.
+COMMAND comprises the program name followed by its arguments, as
+in `make-process'.  Signal `file-error' and emit a warning if
+COMMAND fails.  Obey file handlers based on `default-directory'."
+  (let ((stderr (make-temp-file "counsel-call-stderr-")))
     (unwind-protect
          (with-temp-buffer
-           (let ((res (apply #'call-process (car program-and-args)
-                             nil (list t stderr-file)
-                             nil (cdr program-and-args))))
-             (if (= 0 res)
-                 (buffer-substring (point-min)
-                                   (- (point-max)
-                                      (if (= ?\n (char-before (point-max)))
-                                          1
-                                        0)))
-               (error "%S exited with %d\n%s"
-                      (mapconcat #'identity program-and-args " ")
-                      res
-                      (with-temp-buffer
-                        (insert-file-contents stderr-file)
-                        (buffer-string))))))
-      (delete-file stderr-file))))
+           (let ((res (apply #'process-file (car command) nil
+                             (list t stderr) nil (cdr command))))
+             (unless (eq res 0)
+               (setq res (list 'file-error
+                               (mapconcat #'identity `(,@command "failed") " ")
+                               res))
+               (lwarn 'ivy :warning "%s"
```

Comment:
I think it's an `:error` (it already failed and we `raise`).

Reply:
The process indeed encountered an error, but that doesn' necessarily translate as an error with Ivy. I'm not disagreeing with you, I'm just playing devil's advocate. I'd be happy with any one of `:debug`, `:warning`, or `:error`. The only thing I'm concerned about is that, the higher the severity, the higher the chance of annoying users, e.g. with false positives.

Here's the description of the different severity levels, for posterity:
```
‘:emergency’
     A problem that will seriously impair Emacs operation soon if you do
     not attend to it promptly.
‘:error’
     A report of data or circumstances that are inherently wrong.
‘:warning’
     A report of data or circumstances that are not inherently wrong,
     but raise suspicion of a possible problem.
‘:debug’
     A report of information that may be useful if you are debugging.
```

I'll let someone else decide.